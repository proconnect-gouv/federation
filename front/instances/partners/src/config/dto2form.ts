import type { DTO2FormConfig } from '@fc/dto2form';

export const DTO2Form: DTO2FormConfig = {
  showFieldValidationMessage: false,
  validateOnFieldChange: false,
  validateOnSubmit: false,
};
