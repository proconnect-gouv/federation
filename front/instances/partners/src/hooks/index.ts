export * from './instance-create';
export * from './instance-update';
export * from './instances';
