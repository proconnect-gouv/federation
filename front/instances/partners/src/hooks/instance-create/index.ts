export * from './instance-create.hook';
