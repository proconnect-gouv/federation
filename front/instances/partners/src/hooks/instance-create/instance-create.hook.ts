import isEmpty from 'lodash.isempty';
import { useCallback } from 'react';
import { useNavigate, useRouteLoaderData } from 'react-router-dom';

import { InstancesService } from '@fc/core-partners';
import type { JSONFieldType } from '@fc/dto2form';
import { UNKNOWN_FORM_ERROR } from '@fc/forms';
import type { HttpClientDataInterface } from '@fc/http-client';

import { RouteLoaderDataIds, SubmitTypes, SubmitTypesMessage } from '../../enums';

export const useInstanceCreate = () => {
  const navigate = useNavigate();
  const schema = useRouteLoaderData(RouteLoaderDataIds.VERSION_SCHEMA) as JSONFieldType[];

  const submitHandler = useCallback(
    async (data: HttpClientDataInterface) => {
      const response = await InstancesService.create(data);

      const hasSubmissionErrors = !response || !isEmpty(response.payload);
      if (hasSubmissionErrors) {
        return (response && response.payload) || UNKNOWN_FORM_ERROR;
      }

      const submitState = {
        message: SubmitTypesMessage.INSTANCE_SUCCESS_CREATE,
        type: SubmitTypes.SUCCESS,
      };

      navigate('..', { replace: true, state: { submitState } });
      return null;
    },
    [navigate],
  );

  return { schema, submitHandler };
};
