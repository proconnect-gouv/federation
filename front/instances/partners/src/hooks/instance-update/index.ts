export * from './instance-update.hook';
