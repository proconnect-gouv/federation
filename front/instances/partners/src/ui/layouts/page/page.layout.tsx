import React from 'react';
import { Outlet } from 'react-router-dom';

export const PageLayout = React.memo(() => (
  <main className="fr-container fr-py-8v">
    <div className="fr-grid-row fr-grid-row--center">
      <Outlet />
    </div>
  </main>
));

PageLayout.displayName = 'PageLayout';
