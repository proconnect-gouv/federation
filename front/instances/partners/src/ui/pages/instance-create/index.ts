export * from './instance-create.page';
