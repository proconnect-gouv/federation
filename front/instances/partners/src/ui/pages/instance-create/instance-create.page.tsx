import React from 'react';

import { DTO2FormComponent } from '@fc/dto2form';
import { t } from '@fc/i18n';

import { useInstanceCreate } from '../../../hooks';

export const InstanceCreatePage = React.memo(() => {
  const { initialValues, schema, submitHandler } = useInstanceCreate();

  return (
    <React.Fragment>
      <h1 className="fr-col-12 fr-col-md-8">{t('Partners.createpage.title')}</h1>
      <div className="fr-col-12 fr-col-md-8 fr-py-6w fc-grey-border--full">
        <div className="fr-col-offset-1 fr-col-10">
          <DTO2FormComponent
            config={{ id: 'DTO2Form-instance-create' }}
            initialValues={initialValues}
            schema={schema}
            onSubmit={submitHandler}
          />
        </div>
      </div>
    </React.Fragment>
  );
});

InstanceCreatePage.displayName = 'InstanceCreatePage';
