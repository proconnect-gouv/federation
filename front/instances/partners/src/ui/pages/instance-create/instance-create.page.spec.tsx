import { render } from '@testing-library/react';

import type { SchemaFieldType } from '@fc/dto2form';
import { DTO2FormComponent } from '@fc/dto2form';
import { t } from '@fc/i18n';

import { useInstanceCreate } from '../../../hooks';
import { InstanceCreatePage } from './instance-create.page';

jest.mock('../../../hooks/instance-create/instance-create.hook');

describe('InstanceCreatePage', () => {
  it('should match the snapshot', () => {
    // Given
    const submitHandlerMock = jest.fn();
    const schemaMock = Symbol('any-schema-mock') as unknown as SchemaFieldType[];
    const initialValuesMock = Symbol('any-initial-values-mock') as unknown as Record<
      string,
      string | string[]
    >;

    jest.mocked(useInstanceCreate).mockReturnValueOnce({
      initialValues: initialValuesMock,
      schema: schemaMock,
      submitHandler: submitHandlerMock,
    });

    jest.mocked(t).mockReturnValueOnce('Partners-form-createTitle-mock');

    // When
    const { container, getByText } = render(<InstanceCreatePage />);
    const titleElt = getByText('Partners-form-createTitle-mock');

    // Then
    expect(container).toMatchSnapshot();
    expect(titleElt).toBeInTheDocument();
    expect(t).toHaveBeenCalledOnce();
    expect(t).toHaveBeenCalledWith('Partners.createpage.title');
    expect(DTO2FormComponent).toHaveBeenCalledOnce();
    expect(DTO2FormComponent).toHaveBeenCalledWith(
      {
        config: { id: 'DTO2Form-instance-create' },
        initialValues: initialValuesMock,
        onSubmit: submitHandlerMock,
        schema: schemaMock,
      },
      {},
    );
  });
});
