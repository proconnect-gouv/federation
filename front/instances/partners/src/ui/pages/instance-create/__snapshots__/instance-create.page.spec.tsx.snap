// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstanceCreatePage should match the snapshot 1`] = `
<div>
  <h1
    class="fr-col-12 fr-col-md-8"
  >
    Partners-form-createTitle-mock
  </h1>
  <div
    class="fr-col-12 fr-col-md-8 fr-py-6w fc-grey-border--full"
  >
    <div
      class="fr-col-offset-1 fr-col-10"
    >
      <div
        data-mockid="DTO2FormComponent"
      />
    </div>
  </div>
</div>
`;
